### PR TITLE
Haiku fixes

### PR DESCRIPTION
**Pull request by extrowerk**
Various fixes for the build system to get the MPD build with meson on Haiku host

### CODE REVIEW

**Comment by extrowerk** [comment]
With this fixes MPD builds fine,but while it compiles the Haiku output and mixer parts, somehow it they are missing from the binary:
```
Output plugins:
 null fifo pipe openal httpd recorder
```

**Comment by extrowerk** [comment]
RSRC still missing, the gen fix is just my stupidity.

**Comment by extrowerk** [comment]
It seems libmixer_plugins.a have the required Haiku symbols, but the generated executable doesn't.

File: src/haiku/meson.build [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-xres = meson.find_program('xres')
+rc = find_program('rc')
+xres = find_program('xres')
 
 rsrc = custom_target(
   'mpd.rsrc',
   output: 'mpd.rsrc',
   input: 'mpd.rdef',
-  command: [rc, '-o', '@OUTPUT@', '@INPUT@'],
+  command: [rc, '-o -q', '@OUTPUT@', '@INPUT@'],
```
Comment (MaxKellermann) [inline src/haiku/meson.build]: What's this? Unrelated? Does this even work, adding two options separated by a space in one argument?

**Comment by MaxKellermann** [comment]
I've cherry-picked two simple commits, but the others confuse me a bit. Can you please rebase, and fold those patches together which are just fixups for previous unmerged commits?

**Comment by extrowerk** [comment]
Will do.

File: src/net/IPv6Address.hxx [hunk 2]
```diff
@@ -53,7 +53,10 @@ class IPv6Address {
 						   uint16_t c, uint16_t d,
 						   uint16_t e, uint16_t f,
 						   uint16_t g, uint16_t h) noexcept {
-		return {{{
+		return {{
```
Comment (alaviss) [inline src/net/IPv6Address.hxx]:
```suggestion
		return { .s6_addr = {
```

This should be the most portable option as defined in the POSIX specification.

Comment (MaxKellermann) [reply src/net/IPv6Address.hxx]: This isn't valid C++. Designated initializers are C99 (and will maybe come to C++ with C++20).

Comment (MaxKellermann) [reply src/net/IPv6Address.hxx]: This is even more portable, and valid C++14: https://github.com/MusicPlayerDaemon/MPD/commit/1fa99da3c254a6a641bdf4eca35d781281c4c268

**Comment by mmuman** [comment]
Ugh, I just noticed this PR now after redoing it all myself… bleh :^)

**Comment by MaxKellermann** [comment]
So this one is obsolete now? If any pieces are missing, please submit a new PR.

**Pull request closed without merging**